src/types/query.rs

```diff
@@ -0,0 +1,64 @@
+use edn_rs::Serialize;
+
+#[derive(Clone)]
+pub struct Query {
+    find: Vec<String>,
+    whr: Option<Vec<String>>,
+}
+
+impl Query {
+    pub fn find(find: Vec<&str>) -> Self {
+        Self {
+            find: find
+                .into_iter()
+                .map(|f| String::from(f))
+                .collect::<Vec<String>>(),
+            whr: None,
+        }
+    }
+
+    pub fn where_clause(mut self, whr: Vec<&str>) -> Self {
+        self.whr = Some(
+            whr.into_iter()
+                .map(|f| String::from(f))
+                .collect::<Vec<String>>(),
+        );
+        self
+    }
+
+    pub fn build(self) -> Result<Self, String> {
+        if self.whr.is_none() {
+            Err(String::from("Where clause is required"))
+        } else {
+            Ok(self)
+        }
+    }
+}
+
+impl Serialize for Query {
+    fn serialize(self) -> String {
+        let mut q = String::from("{:query\n {:find [");
```

Comment:
You add `\n`s for better readability on printing?
That's probably a pretty good idea for debugging.

Reply:
yes!

Reply:
Noice